scene/resources/resource_format_text.cpp

```diff
@@ -487,6 +492,14 @@ Error ResourceLoaderText::load() {
 		} else {
 			Ref<Resource> res = ResourceLoader::load(path, type);
 
+#ifdef MODULE_GDSCRIPT_ENABLED
+			// GDScript has a way to load cyclic dependencies
+			if (res.is_null() && ResourceLoader::get_resource_type(path) == "GDScript") {
```

Comment:
I know @reduz won't be pleased with this, but I wonder if there's another way to do this? GDScript is pretty much the only resource (with the workaround found for packed scenes in `GDScriptCache`) that supports cyclic resources.

`ResourceLoader::load(path, type)` returns a null `Variant` when it's already loading that resource, instead of returning the resource currently loaded. But it cannot do that because that resource doesn't exist, hence that `GDScript` check.